jaeger-jaxrs2/src/test/java/com/uber/jaeger/propagation/ServerFilterIntegrationTest.java

```diff
@@ -0,0 +1,197 @@
+package com.uber.jaeger.propagation;
+
+import static com.uber.jaeger.Constants.DEBUG_ID_HEADER_KEY;
+import static com.uber.jaeger.Constants.DEBUG_JAEGER_SPAN_BAGGAGE_HEADER;
+import static org.junit.Assert.assertEquals;
+
+import com.uber.jaeger.context.ActiveSpanSourceTraceContext;
+import com.uber.jaeger.context.TraceContext;
+import com.uber.jaeger.filters.jaxrs2.ClientFilter;
+import com.uber.jaeger.filters.jaxrs2.ServerFilter;
+import com.uber.jaeger.reporters.NoopReporter;
+import com.uber.jaeger.samplers.ConstSampler;
+import io.opentracing.Span;
+import io.opentracing.Tracer;
+import java.io.IOException;
+import java.net.URI;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Priorities;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.core.Application;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response;
+import org.glassfish.grizzly.http.server.HttpServer;
+import org.glassfish.jersey.grizzly2.httpserver.GrizzlyHttpServerFactory;
+import org.glassfish.jersey.server.ResourceConfig;
+import org.glassfish.jersey.test.JerseyTest;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class ServerFilterIntegrationTest extends JerseyTest {
+
+  // Base URI the Grizzly HTTP server will listen on
+  private static final String BASE_URI = "http://localhost:8080/";
+
+  private Tracer clientTracer =
+      new com.uber.jaeger.Tracer
+          .Builder("client-tracer", new NoopReporter(), new ConstSampler(true))
+          .build();
+
+  private TraceContext clientTraceContext = new ActiveSpanSourceTraceContext(clientTracer);
+
+  private HttpServer server;
+
+  @Override
+  @Before
+  public void setUp() throws Exception {
+    super.setUp();
+
+    // start the server
+    final ResourceConfig rc = (ResourceConfig) configure();
+    server = GrizzlyHttpServerFactory.createHttpServer(URI.create(BASE_URI), rc);
+  }
+
+  @Override
+  @After
+  public void tearDown() throws Exception {
+    super.tearDown();
+
+    // stop the server
+    if (server.isStarted()) {
+      server.stop();
+    }
+  }
+
+  @Override
+  protected Application configure() {
+
+    io.opentracing.Tracer serverTracer =
+        new com.uber.jaeger.Tracer
+            .Builder("server-tracer", new NoopReporter(), new ConstSampler(true))
+            .build();
+
+    TraceContext serverTraceContext = new ActiveSpanSourceTraceContext(serverTracer);
```

Comment:
`ActiveSpanSourceTraceContext` - this class is going away in the next release, so let's either wait with this PR or make it off the next release branch